lib/workflow/stores/mongo.js

```diff
@@ -52,22 +52,26 @@ function mongoStoreFactory(waterline, Promise, Constants, Errors, assert, _) {
      * @param {Object} indexObject an object with keys that correspond to the mogo
      * collections on which to place indexes and values that are arrays of indexes
      * @example {
-     *          taskdependencies: [
-     *              {taskId: 1},
-     *              {graphId: 1},
-     *              {taskId: 1, graphId: 1}
-     *          ],
-     *          graphobjects: [
-     *              {instanceId: 1}
-     *          ]
+     *       taskdependencies: [
+     *           {index: {taskId: 1}, unique: true},
```

Comment:
Ok, there was a bit of misunderstanding on my part as to the actual bug here. Yes, we do create indexes/enforce uniqueness if we add them to the waterline model, however waterline does not support compound indexes. I've edited to use the waterline models for all simple index/unique necessities and use the mongo method to set the one compound index we want.